### PR TITLE
Fixing infinite spinning animation on user menu ctrl+click

### LINKED ISSUE #616: Ctrl-click on entries in the user menu will keep spinner spinning
When you Ctrl-click on entries in the Apps menu on the top left, the spinner does not show and the menu closes directly – which is the correct behavior.

When you Ctrl-click on entries in the User menu on the top right, the menu stays open and the spinner is shown indefinitely. This should be fixed to behave as the apps menu does. :)

cc @nextcloud/javascript

### PR DESCRIPTION
**Pull request by Faldon**
Resembling the behaviour of the app menu, the user menu now closes on ctrl+click and does not show the loading spinner.

Fixes #616

### CODE REVIEW

File: core/js/js.js [hunk 1]
```diff
@@ -1519,14 +1519,21 @@ function initCore() {
...
+		// show loading feedback
 		$menu.delegate('a', 'click', function(event) {
 			var $page = $(event.target);
 			if (!$page.is('a')) {
 				$page = $page.closest('a');
 			}
-			$page.find('img').remove();
-			$page.find('div').remove(); // prevent odd double-clicks
-			$page.prepend($('<div/>').addClass('icon-loading-small-dark'));
+			if(!event.ctrlKey) {
```
Comment (MorrisJobke) [inline core/js/js.js]: On OS X this is `event.metaKey` could you also check for this?

Comment (MorrisJobke) [inline core/js/js.js]: And also adjust the other menu too :)

**Comment by icewind1991** [comment]
How about middle mousebutton?

**Comment by MorrisJobke** [comment]
> How about middle mousebutton?

On a trackpad there is no middle mousebutton 😝 



@Faldon Thanks! Works really nice now :)

**Comment by MorrisJobke** [comment]
👍

**Comment by schiessle** [comment]
> On a trackpad there is no middle mousebutton

Not everyone uses a trackpad and some people have a well configured track pack, e.g. I can do a middle click by tabbing with three fingers :wink: 

Also it is not a either or... I would love if we could extend the behavior for middle click as well.  @Faldon do you think you can add this? Thanks!

**Comment by Faldon** [comment]
Will look at it tomorrow. ;-)

**Comment by MorrisJobke** [comment]
> Not everyone uses a trackpad and some people have a well configured trackpack, e.g. I can do a middle click by tapping with three fingers 😉

I read it as question directly to me 🙈 Sorry, just misunderstood this one.

**Comment by jancborchardt** [comment]
cc @ChristophWurst here cause of the good timing with same pull request at https://github.com/nextcloud/server/pull/661 ;D We should continue here.

**Comment by Faldon** [comment]
Hmmm... Using Chrome on Arch Linux, I cannot even detect the middle mouse click.
I get the event.which === 1 for left button click.
When I click the menu with middle mouse button, no spinner appears and menu does not close.

**Comment by Faldon** [comment]
So, did some more tests:
It appears, that the click event is not raised on middle button and right button click. At least not in Chrome  on Arch and Fedora.
For testing, I set the event to prevent the default immediately after the event occurs. While this successfully stopped navigating on left button click (and decorating every menu entry with spinning icons :-D ), on middle button click my browser just opened the page in a new tab. No event handling after all.

**Comment by MorrisJobke** [comment]
Let's get this in then :)

@juliushaertl @ChristophWurst Mind to review?

**Comment by jancborchardt** [comment]
@Faldon any update here? :)

**Comment by Faldon** [comment]
Sorry, was a little busy.
With the changes @ChristophWurst proposed, I can reproduce, that middle-clicking and cttl-/meta-clicking does not open the links anymore (though it closes the menu now.)

But I think I have got a proper solution:

```
@@ -1520,12 +1523,12 @@
 		var $menu = $('#header #settings');
 
 		// show loading feedback
-		$menu.delegate('a', 'click', function(event) {
+		$menu.delegate('a', 'mousedown', function(event) {
 			var $page = $(event.target);
 			if (!$page.is('a')) {
 				$page = $page.closest('a');
 			}
-			if(!event.ctrlKey && !event.metaKey) {
+			if(event.which === 1 && !event.ctrlKey && !event.metaKey) {
 				$page.find('img').remove();
 				$page.find('div').remove(); // prevent odd double-clicks
 				$page.prepend($('<div/>').addClass('icon-loading-small-dark'));
@@ -1533,7 +1536,12 @@
 				// Close navigation when opening menu entry in
 				// a new tab
 				OC.hideMenus();
+				// On middle click or on first button click with ctrl key or meta key hold
+				if(event.which === 2 || (event.which === 1 && (event.ctrlKey || event.metaKey))) {
+					window.open($page, '_blank');
-			}
+				}
+			}
+			$($page).click();
 		});
 	}
```

It's a little bit more logic, but handles opening links only if the primary button was clicked, with or without ctrl key or meta key modifier, or if the middle mouse button was clicked.
I will push it for testing (I have a mac keyboard but it is attached to Arch Linux. :-) )

**Comment by juliushaertl** [comment]
Code looks good and works fine for me, please check for the failing drone build. @Faldon

**Comment by Faldon** [comment]
@juliushaertl I have to confess, I'm somewhat stuck with the drone output. I can determine, there is one failed feature test in sharing-v1 regarding permissions not registered in OC_Filesystem.

**Comment by MorrisJobke** [comment]
It's weird ... this has failed on others PRs too. :(

@rullzer Do you have any idea how this works?

**Comment by rullzer** [comment]
See: https://github.com/nextcloud/server/pull/752

**Comment by juliushaertl** [comment]
#752 is merged, @Faldon could you rebase your pullrequest?

**Comment by Faldon** [comment]
I hope this was right -.-

**Comment by MorrisJobke** [comment]
> I hope this was right -.-

Doesn't seem so. Should I create a new PR with only your changes?

**Comment by MorrisJobke** [comment]
I tried to resolve this here: #778

**Comment by jancborchardt** [comment]
Closing in favor of that new PR then.

**Pull request closed without merging**